src/gui/filedetails/sharemodel.cpp

```diff
@@ -766,6 +767,33 @@ void ShareModel::slotShareExpireDateSet(const QString &shareId)
     Q_EMIT dataChanged(shareModelIndex, shareModelIndex, { ExpireDateEnabledRole, ExpireDateRole });
 }
 
+void ShareModel::slotDeleteE2EeShare(const SharePtr &share) const
```

Comment:
are you sure this is really a const method ?
I see that we call `emit serverError(code, message);`
I do not think signals are const methods. So we are kind of abusing the const qualifier (via the lambda in the connect)